### PR TITLE
Add level builder option for randomly rotating Maze puzzles

### PR DESCRIPTION
**Pull request by joshlory**
screen shot 2017-06-01 at 3 26 24 pm

### CODE REVIEW

File: apps/src/appMain.js [hunk 1]
```diff
@@ -62,6 +62,13 @@ export default function (app, levels, options) {
 
   studioApp().configure(options);
 
+  if (options.level && options.level.shapeShift) {
+    const rotate = m => m[0].map((x, i) => m.map(x => x[m.length - i - 1]));
+    for (let i = 0; i < Math.random() * 4; i++) {
```
Comment (balderdash) [inline apps/src/appMain.js]: Nit: This won't give you an even distribution of rotations, and you probably want it to do 0-3 rotations, rather than 1-4, so something like `for (let i = 1, max = Math.random() * 4; i < max; i++) { ... }`

File: apps/src/appMain.js [hunk 2]
```diff
@@ -62,6 +62,13 @@ export default function (app, levels, options) {
...
+      options.level.map = rotate(options.level.map);
+      options.level.startDirection = (options.level.startDirection + 3) % 4;
+    }
+  }
```
Comment (Hamms) [inline apps/src/appMain.js]: why is this here and not in Maze's loadLevel?

File: apps/src/appMain.js [hunk 3]
```diff
@@ -62,6 +62,13 @@ export default function (app, levels, options) {
 
   studioApp().configure(options);
 
+  if (options.level && options.level.shapeShift) {
+    const rotate = m => m[0].map((x, i) => m.map(x => x[m.length - i - 1]));
```
Comment (balderdash) [inline apps/src/appMain.js]: The `x` in `(x, i)` is unused, right? Maybe rename that to `_`.

File: apps/src/appMain.js (hunk 1 above)
Comment (joshlory) [reply apps/src/appMain.js]: Good catch!  I totally missed that `Math.random()` will regenerate on each iteration.

File: apps/src/appMain.js (hunk 2 above)
Comment (joshlory) [reply apps/src/appMain.js]: Right, sorry, will move it there.

File: apps/src/appMain.js (hunk 3 above)
Comment (joshlory) [reply apps/src/appMain.js]: The index gets used inside the second `map`.

**Comment by joshlory** [comment]
@Hamms PTAL
